src/blueprints/Do.Blueprints.Service.Application/ServiceSpecExtensions.cs

```diff
@@ -3,19 +3,26 @@
 using Microsoft.Extensions.DependencyInjection;
 using Shouldly;
 using System.Reflection;
+using System.Text.RegularExpressions;
 
 namespace Do;
 
 public static class ServiceSpecExtensions
 {
     #region MockOverrider
 
+    public static T The<T>(this Searcher getMe, params object?[] mockOverrides) where T : notnull =>
+        getMe.Spec.GiveMe.The<T>(mockOverrides);
     public static T The<T>(this Stubber _, params object?[] mockOverrides) where T : notnull =>
         ServiceSpec.ServiceProvider.OverrideMocksAndGetRequiredService<T>(mockOverrides);
 
+    public static T An<T>(this Searcher getMe, params object?[] mockOverrides) where T : notnull =>
+        getMe.Spec.GiveMe.An<T>(mockOverrides);
     public static T An<T>(this Stubber _, params object?[] mockOverrides) where T : notnull =>
         ServiceSpec.ServiceProvider.OverrideMocksAndGetRequiredService<T>(mockOverrides);
 
+    public static T A<T>(this Searcher getMe, params object?[] mockOverrides) where T : notnull =>
+        getMe.Spec.GiveMe.A<T>(mockOverrides);

```

Comment:
Searcher'ı veri tabanına yazılmış bir entity'yi belirli kriterlere göre alalım , o yüzden `The<>` olmalı extension diye düşünmüştük, `A<>` veya `An<>` dersek herhangi bir manasına da gelebiliyor, zaten o zaman da mocklarız, gerek olmaz diye düşünüyoum

Reply:
`ServiceSpec` zaten elimizde var oradan ServiceProvider'a ulaşabiliriz, `GiveMe`'den almaya gerek yok